### PR TITLE
action#6440 - dynamic UI for test cloning

### PR DESCRIPTION
**Pull request by aaannz**
- enhance scheduled_table datatable in similar way as results are
- add on-click action to reload datatable in case of cloning, prio change and cancel
  + this is stupid since it download all data, but only for scheduled table so reasonable amount I guess
- expand list_ajax to include job state option
- add url string params for test overview, cancel and prio adjustment

### CODE REVIEW

File: public/javascripts/tests.js [hunk 1]
```diff
@@ -84,17 +100,105 @@ function renderTestsList(jobs, is_operator, restart_url) {
...
+            },
+            { targets: 1,
+                className: "test",
+                "render": function ( data, type, row ) {
+                    if (type === 'display') {
+                        var url = test_url.replace('REPLACEIT', row['id']);
+                        return '<a class="overview_' + row['result'] + '" href="' + url + '">' + data + '</a>';
+                    } else {
+                        return data;
+                    }
```
Comment (coolo) [inline public/javascripts/tests.js]: these functions should be extracted I'd say - they look pretty much the same for the 2 tables.

**Comment by coolo** [comment]
be careful with the phrase responsive UI - it kind of sets expectations :)
  http://en.wikipedia.org/wiki/Responsive_web_design

**Comment by ancorgs** [comment]
Yeah, I was quite disappointed when I found what @aaannz meant with responsive :)

**Comment by aaannz** [comment]
Ok, renamed to dynamic, what that stands for?
Updated some testsuites, but still fails because I was unable to log in using phantomjs.

File: public/javascripts/tests.js [hunk 2]
```diff
@@ -84,17 +100,109 @@ function renderTestsList(jobs, is_operator, restart_url) {
...
+        "ajax": {
+            "url": "/tests/list_ajax",
+            "type": "POST", // we use POST as the URLs can get long
+            "data": function(d) {
+            var ret = {
+                "relevant": $('#relevantfilter').prop('checked'),
+                "state" : 'scheduled'
+            };
+            if (jobs != null) {
+                ret['jobs'] = jobs;
```
Comment (coolo) [inline public/javascripts/tests.js]: this doesn't make sense - the jobs array are dones.

Comment (aaannz) [inline public/javascripts/tests.js]: I added a param 'state' so I can choose if done or scheduled

Comment (aaannz) [inline public/javascripts/tests.js]: Hmm.. as I look into it, I think I added 'state' param. It works in tests, but maybe not as I imagined. Will need to check.

**Comment by coolo** [comment]
I was thinking about this a bit more and I think you're running the wrong direction. If you fill two tables with 2 independent ajax calls (in an app having just one thread) you get ugly flickering. And you seem to  reload the full table when you change something. This will lead to usability desaster as you get a quite huge delay after the click. So if you want to restart several jobs you end up clicking on the wrong thing as you get a neat queue of ajax calls done - they don't block

**Comment by aaannz** [comment]
I didn't want to use one call, well for one I don't know how to do it yet, but mainly for the amount of data it would transfer. True I didn't count with possible ajax call delay.
Btw. I don't reload results table, only scheduled so restarting should be ok, canceling job might by and issue.

So instead of table reload just create entries by hand? That doesn't seem right to me.

**Comment by aaannz** [comment]
I'll open new PR. I don't want to link this discussion to new code.

**Pull request closed without merging**